### PR TITLE
added a simple model controller plugin that uses a twist message

### PR DESCRIPTION
**Pull request by piyushk**
This controller can be used to move any entity on the XY plane as long as it has a high enough inertia. I use it when I have to spawn a large number of robots in simulation. I give the robots a cylindrical collision model, high inertia with a center of mass located close to the ground. This controller gives a fair amount of speedup in comparison to simulating the complete model and a differential drive controller.

### CODE REVIEW

**Comment by piyushk** [comment]
I'll reopen this in a second. I forgot to test odometry publication.

**Comment by davetcoleman** [comment]
@gerkey this sounds like player/stage functionality, can you weight in on adding this new plugin?

@nkoenig

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_model_controller.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+/*
+ * Desc: Simple model controller that uses a twist message to move a robot on
+ *       the xy plane.
+ * Author: Piyush Khandelwal
+ * Date: 29 July 2013
+ */
+
+#ifndef GAZEBO_ROS_MODEL_CONTROLLER_5DRC9ADD
+#define GAZEBO_ROS_MODEL_CONTROLLER_5DRC9ADD
```
Comment (davetcoleman) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_model_controller.h]: What is 5DRC9ADD ?  Typically you use the pkg name then file name.

Comment (piyushk) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_model_controller.h]: I can change it. I have a script that adds the anonymizer.

**Comment by piyushk** [comment]
I've fixed some problems and reopened the request. Let me know what you guys think about adding this plugin. I do use it frequently in gazebo as an alternate robot controller, and for moving entities which are not dynamically suited to move around in space (I have external human controlled observers in the simulation environment).

Thanks!

**Comment by nkoenig** [comment]
The plugin name should convey its functionality. Can this plugin be renamed to something like: gazebo_ros_planar_move?

**Comment by davetcoleman** [comment]
+1

**Comment by piyushk** [comment]
Done.

**Comment by davetcoleman** [comment]
Can you first write the description and xml example on [this](http://gazebosim.org/wiki/Tutorials/1.9/ROS_Motor_and_Sensor_Plugins) page so I can try it out?

Thanks again for doing this!

**Comment by piyushk** [comment]
I have documented this here: http://gazebosim.org/wiki/Tutorials/1.9/ROS_Motor_and_Sensor_Plugins#Planar_Move_Plugin

I have written a sample URDF, but I cannot test it right now. Let me know if something does not work.

**Pull request merged**